src/System.Private.Uri/src/System/DomainNameHelper.cs

```diff
@@ -324,7 +324,7 @@ internal unsafe static string IdnEquivalent(char* hostname, int start, int end,
             {
                 // just lowercase for ascii
                 string unescapedHostname = new string(hostname, start, end - start);
-                return ((unescapedHostname != null) ? unescapedHostname.ToLowerInvariant() : null);
+                return unescapedHostname.ToLowerInvariant();
```

Comment:
A constructor will never return null.  If it did, that would be a giant CLR bug.